tests/integration/galley/webhook_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package galley
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+)
+
+var (
+	vwcName    = "istio-galley"
+	deployName = "istio-galley"
+	istioNs    = "istio-system"
+	sleepDelay = 3 * time.Second // How long to wait to give the reconcile loop an opportunity to act
+)
+
+var removeGalleyYaml = fmt.Sprintf(`
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: %s
+  namespace: %s
+`, deployName, istioNs)
+
+// Verify that scaling up/down doesn't modify webhook configuration
+func TestWebhookScaling(t *testing.T) {
+	framework.NewTest(t).
+		// Limit to Kube environment as we're testing integration of webhook with K8s.
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			env := ctx.Environment().(*kube.Environment)
+			startGen := getVwcGeneration(vwcName, t, env)
+
+			// Scale up
+			scaleDeployment(istioNs, deployName, 2, t, env)
+			// Wait a bit to give the ValidatingWebhookConfiguration reconcile loop an opportunity to act
+			time.Sleep(sleepDelay)
+			gen := getVwcGeneration(vwcName, t, env)
+			if gen != startGen {
+				t.Fatalf("ValidatingWebhookConfiguration was updated unexpectedly on scale up")
+			}
+
+			// Scale down
+			scaleDeployment(istioNs, deployName, 1, t, env)
+			// Wait a bit to give the ValidatingWebhookConfiguration reconcile loop an opportunity to act
+			time.Sleep(sleepDelay)
+			gen = getVwcGeneration(vwcName, t, env)
+			if gen != startGen {
+				t.Fatalf("ValidatingWebhookConfiguration was updated unexpectedly on scale down")
+			}
+		})
+}
+
+// Verify that removing the galley deployment results in the webhook configuration being removed
+func TestWebhookRemoved(t *testing.T) {
+	framework.NewTest(t).
+		// Limit to Kube environment as we're testing integration of webhook with K8s.
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			env := ctx.Environment().(*kube.Environment)
+
+			// Remove galley webhook
+			env.DeleteContents(istioNs, removeGalleyYaml)
```

Comment:
Please use a strongly typed Accessor method to do this. If one doesn't exist, you can add one.

Reply:
if you are going to be tampering with the istio installation please add a `label.CustomSetup` to the test. We have tests that run with no setup mode against an existing install that this will break

Reply:
@howardjohn This label applies at the suite level, correct? Does that mean these tests should also be moved into their own suite, so we don't change the labeling of the existing galley tests?

Reply:
I thought you could apply to just a test as well? maybe not though

Reply:
@ozevren To clarify, do you mean something like env.DeleteDeployment? Or did you have something else in mind?

Reply:
@howardjohn It appears to be suite level, which is fine... it looks like breaking these tests out into one or more new suites makes more sense than leaving them attached to the existing galley suite.

Reply:
Something like Accessor.DeleteDeployment. The methods on Environment & Accessor should be mostly generic utility methods. They can end up doing complicated things (i.e. select a bunch of pods and wait for them to startup) but they should not do any Istio specific operations. Those should be done at the component level.

Reply:
@howardjohn 
You can apply label to tests, but these types of tests should work when running via `go test` as well as in a more generic "run all tests" mode in automation. Do you have a plan to resolve this?